signer/signer.go

```diff
@@ -90,8 +90,9 @@ func (s *genericSigner) Sign(ctx context.Context, desc ocispec.Descriptor, opts
 		SigningAgent:  signingAgent, // TODO: include external signing plugin's name and version. https://github.com/notaryproject/notation-go/issues/80
 	}
 
-	if !opts.Expiry.IsZero() {
-		signReq.Expiry = opts.Expiry
+	// Add expiry only if ExpiryDuration is not nil and its value is not zero
+	if opts.ExpiryDuration != nil && opts.ExpiryDuration.Seconds() != 0 {
+		signReq.Expiry = signReq.SigningTime.Add(*opts.ExpiryDuration)
 	}
```

Comment:
If we don't add expiry if its value is zero, then why we need to make `opts.ExpiryDuration` a `*time.Duration` pointer? Will `ExpiryDuration time.Duration` be better?

Reply:
Having expiry as pointer(`*time.Duration`) helps to convey message that expiry is optional, without need of any documentation. Having it as `time.Duration` means now caller/user will need to read documentation to understand how will it behaves.

Reply:
In the golang conversion, `time.Duration` is more preferable than `*time.Duration`. As a golang library / module, it is better to be consistent with the built-in libraries.

Reply:
Additionally, for library code, taking `*time.Duration` may cause usability issue. 

For instance,
```go
const expiryForCertianApp time.Duration = 30 * 24 * time.Hour // one month

signOpts := notation.SignOptions{
	ArtifactReference:  opts.reference,
	SignatureMediaType: mediaType,
	ExpiryDuration:     &expiryForCertianApp,
	PluginConfig:       pluginConfig,
}
```
The above code will have compile error since you cannot take an address of a const. However, constant expiry duration is common.

Instead of the above code, developers have to do
```go
expriyDuration := expiryForCertianApp
signOpts := notation.SignOptions{
	ArtifactReference:  opts.reference,
	SignatureMediaType: mediaType,
	ExpiryDuration:     &expriyDuration,
	PluginConfig:       pluginConfig,
}
```
which does not sound right.

Reply:
I have seen golang code use pointers for time.Duration in case where absence of value has a meaning ([ref1]( https://github.com/grpc/grpc-go/blob/v1.51.0/internal/serviceconfig/serviceconfig.go#L139-L145), [ref2](https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xdsresource/type_rds.go#L137), [ref3](https://github.com/kubernetes-sigs/controller-runtime/blob/c83076e9f79258f23bcb200cf2391cf4ade939cf/pkg/config/v1alpha1/types.go#L56)). Yes, there is some inconvenience while writing code but having cleaner API/method definition will reduce potential bugs.

Reply:
As per discussion updated PR to use `time.Duration`